src/OpenFeature/Model/ImmutableMetadata.cs

```diff
@@ -45,7 +45,13 @@ public ImmutableMetadata(Dictionary<string, object> metadata)
     /// <returns>The integer value associated with the key, or null if the key is not found.</returns>
```

Comment:
The behavior will need to be updated in the document comments for both the methods.